lib/collection/response.js

```diff
@@ -107,6 +107,8 @@ _.inherit((
 
 _.assign(Response.prototype, /** @lends Response.prototype */ {
     update: function (options) {
+        // options.stream accepts new Buffer() as well as new Buffer().toJSON()
+        options.stream && options.stream.type === 'Buffer' && (options.stream = new Buffer(options.stream));
```

Comment:
Newer node versions use `Buffer.from()`, `new Buffer()` is deprecated, so can you handle that case? I think @kunagpal has handled it somewhere already.

Also, is `Buffer` available in browsers? This code is cross platform, so it should work in browsers as well.